### PR TITLE
[T5] Add training documenation 

### LINKED ISSUE #3333: Finetuning T5 Model
Hi there. I am trying to fine tune T5, but I have noticed your documentation gives conflicting instructions.

In modeling_t5.py you say
> To match pre-training, T5 input sequence should be formatted with [CLS] and [SEP] tokens

but then in tokenization_t5.py both of those tokens are set to None and the only tokens that are defined are EOS, UNK, and PAD.
Additionally, the actual T5 implementation makes no mention of SEP and CLS as far as I can tell.
Given that, could you clarify how we should be formatting our training data for HuggingFace's T5 implementation? Thank you!

### PR DESCRIPTION
**Pull request by patrickvonplaten**
- Fixes T5 docstring regarding pretraining
- Add detailed description on how to process input and target for T5 training

### CODE REVIEW

File: docs/source/model_doc/t5.rst [hunk 1]
```diff
@@ -16,6 +16,37 @@ To facilitate future work on transfer learning for NLP, we release our dataset,
 
 The Authors' code can be found `here <https://github.com/google-research/text-to-text-transfer-transformer>`_ .
 
+Training
+~~~~~~~~~~~~~~~~~~~~
```
Comment (patrickvonplaten) [inline docs/source/model_doc/t5.rst]: @thomwolf - Could you look over it and check? :-)

File: docs/source/model_doc/t5.rst [hunk 2]
```diff
@@ -16,6 +16,37 @@ To facilitate future work on transfer learning for NLP, we release our dataset,
...
+  
+::
+
+  input_ids = tokenizer.encode('The house is wonderful. </s>', return_tensors='pt')
+  decoder_input_ids = tokenizer.encode('<pad> Das Haus ist wunderbar.', return_tensors='pt')
+  lm_labels = tokenizer.encode('Das Haus ist wunderbar. </s>', return_tensors='pt')
+  model(input_ids=input_ids, decoder_input_ids=decoder_input_ids, lm_labels=lm_labels)
+
 Tips
 ~~~~~~~~~~~~~~~~~~~~
```
Comment (patrickvonplaten) [inline docs/source/model_doc/t5.rst]: @thomwolf - Could you look over it and check? :-)

File: src/transformers/modeling_t5.py [hunk 3]
```diff
@@ -699,32 +699,24 @@ def forward(
     Args:
         input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`):
             Indices of input sequence tokens in the vocabulary.
```
Comment (patrickvonplaten) [inline src/transformers/modeling_t5.py]: I think the example below is not correct for T5 - deleted it @thomwolf

File: docs/source/model_doc/t5.rst [hunk 4]
```diff
@@ -16,6 +16,37 @@ To facilitate future work on transfer learning for NLP, we release our dataset,
...
+  model(input_ids=input_ids, decoder_input_ids=decoder_input_ids, lm_labels=lm_labels)
+
+- Supervised training
+  In this setup the input sequence and output sequence are standart sequence to sequence input output mapping.
+  In translation, *e.g.* the input sequence "The house is wonderful." and output sequence "Das Haus ist wunderbar." should 
+  be processed as follows:
+  
+::
+
+  input_ids = tokenizer.encode('The house is wonderful. </s>', return_tensors='pt')
```
Comment (craffel) [inline docs/source/model_doc/t5.rst]: Technically T5 uses a task prefix to distinguish between different tasks; for En-De translation we'd use `translate English to German: The house is wonderful.`

**Pull request merged**